File: src/test/java/kitchenpos/domain/FakeOrderRepository.java
package kitchenpos.domain;

import kitchenpos.order.domain.OrderRepository;
import kitchenpos.order.domain.Order;
import kitchenpos.order.domain.OrderTable;
import kitchenpos.order.domain.OrderStatus;

import java.util.*;

public class FakeOrderRepository implements OrderRepository {

    private Map<UUID, Order> inMemory = new HashMap<>();

    @Override
    public Optional<Order> findById(UUID orderId) {
        return Optional.ofNullable(inMemory.get(orderId));
    }

    @Override
    public List<Order> findAll() {
        return inMemory.values().stream().toList();
    }

    @Override
    public Order save(Order order) {
        inMemory.put(order.getId(), order);
        return order;
    }

    @Override
    public boolean existsByOrderTableAndStatusNot(OrderTable orderTable, OrderStatus orderStatus) {
        return inMemory.values()
                .stream()
                .anyMatch(order -> order.getOrderTable().equals(orderTable) && !order.getStatus().equals(orderStatus));
    }
}

File: src/test/java/kitchenpos/testsupport/FakeOrderRepository.kt
package kitchenpos.testsupport

import java.util.Optional
import java.util.UUID
import java.util.concurrent.ConcurrentHashMap
import kitchenpos.eatinorders.domain.Order
import kitchenpos.eatinorders.domain.OrderRepository
import kitchenpos.eatinorders.domain.OrderStatus
import kitchenpos.eatinorders.domain.OrderTable

class FakeOrderRepository : OrderRepository {
    private val storage: MutableMap<UUID, Order> = ConcurrentHashMap()

    override fun save(order: Order): Order {
        storage[order.id] = order
        return order
    }

    override fun findById(orderId: UUID): Optional<Order> {
        return Optional.ofNullable(storage[orderId])
    }

    override fun existsByOrderTableAndStatusNot(
        orderTable: OrderTable,
        orderStatus: OrderStatus
    ): Boolean {
        return storage.values
            .any {
                it.orderTable.id == orderTable.id &&
                        it.status == orderStatus
            }
    }

    override fun findAll(): MutableList<Order> {
        return storage.values.toMutableList()
    }
}